modules/aruco/src/dictionary.cpp

```diff
@@ -94,27 +94,46 @@ static inline bool readParameter(const FileNode& node, T& parameter)
     return false;
 }
 
-bool Dictionary::readDictionary(const cv::FileNode& fn, cv::Ptr<cv::aruco::Dictionary> &dictionary)
+bool Dictionary::readDictionary(const cv::FileNode& fn)
 {
-    int nMarkers = 0, markerSize = 0;
-    if(fn.empty() || !readParameter(fn["nmarkers"], nMarkers) || !readParameter(fn["markersize"], markerSize))
+    int nMarkers = 0, _markerSize = 0;
+    if (fn.empty() || !readParameter(fn["nmarkers"], nMarkers) || !readParameter(fn["markersize"], _markerSize))
         return false;
-    cv::Mat bytes(0, 0, CV_8UC1), marker(markerSize, markerSize, CV_8UC1);
+    Mat bytes(0, 0, CV_8UC1), marker(_markerSize, _markerSize, CV_8UC1);
     std::string markerString;
     for (int i = 0; i < nMarkers; i++) {
         std::ostringstream ostr;
         ostr << i;
         if (!readParameter(fn["marker_" + ostr.str()], markerString))
             return false;
-
         for (int j = 0; j < (int) markerString.size(); j++)
             marker.at<unsigned char>(j) = (markerString[j] == '0') ? 0 : 1;
-        bytes.push_back(cv::aruco::Dictionary::getByteListFromBits(marker));
+        bytes.push_back(Dictionary::getByteListFromBits(marker));
     }
-    dictionary = cv::makePtr<cv::aruco::Dictionary>(bytes, markerSize);
+    int _maxCorrectionBits = 0;
+    readParameter(fn["maxCorrectionBits"], _maxCorrectionBits);
+    *this = Dictionary(bytes, _markerSize, _maxCorrectionBits);
     return true;
 }
 
+void Dictionary::writeDictionary(const String& outputDictFile) {
```

Comment:
I propose to add simple write->read test to check the new API. It could be done in python to cover python bindings too.

Reply:
add .py test_write_read_dict